src/lib/dnssd/Discovery_ImplPlatform.cpp

```diff
@@ -75,8 +85,9 @@ static void HandleNodeBrowse(void * context, DnssdService * services, size_t ser
 
         auto & ipAddress = services[i].mAddress;
 
-        // Check if SRV, TXT and AAAA records were received in DNS responses
-        if (strlen(services[i].mHostName) == 0 || services[i].mTextEntrySize == 0 || !ipAddress.has_value())
+        // Check that this is not operational browse result and if SRV, TXT and AAAA records were received in DNS responses
+        if (strncmp(services[i].mType, kOperationalServiceName, sizeof(kOperationalServiceName)) != 0 &&
```

Comment:
This may cause that null `ipAddress` is dereferenced if the address has not been returned for the browse query.

Reply:
This is filled into OperationalNodeBrowseData for operational which doesn't have any member for ip address, hence for operational browse ipAddress is not needed or accessed.

Reply:
I understand but the `else` code assumes that the address exists so by adding this code you make it possible to dereference a null optional.

Reply:
This line:
```
Inet::IPAddress * address = &(*ipAddress);
```
will have undefined behavior (not necessarily crash, since `operator*` does not do checks) `ipAddress` has no value.

The logic here needs to be revamped, and the assumptions of `HandleNodeResolve` need to be audited.

Alternately, we should have two different versions of `HandleNodeResolve` for operational and commissionable, since they seem to have quite different logic anyway.

Reply:
fixed it.